### ISSUE TITLE
[Build] incomplete git updates after pulling most recent master branch

### ISSUE THREAD

**Issue by willyborn** [arrayfire/arrayfire#3098]
pulling the new master, results in a few errors.

Description
===========
When pulling the new master version of arrayfire (3.9), git displayed a few errors during the update.  Later this produced compilation errors in src/backend/cpu/threads/* resulting a failed afcpu.dll.  Manually deleting the mentioned folders, did solve the problem.

I assume that the individual files have to be deleted before the folder is deleted.

Error Log
---------
```
$ git pull upstream master
From https://github.com/arrayfire/arrayfire
 * branch              master     -> FETCH_HEAD
warning: unable to rmdir 'assets': Directory not empty
warning: unable to rmdir 'extern/cub': Directory not empty
warning: unable to rmdir 'extern/forge': Directory not empty
warning: unable to rmdir 'extern/glad': Directory not empty
warning: unable to rmdir 'extern/spdlog': Directory not empty
warning: unable to rmdir 'src/backend/cpu/threads': Directory not empty
warning: unable to rmdir 'test/data': Directory not empty
warning: unable to rmdir 'test/gtest': Directory not empty
Updating d86edd18..f6ed89cb
Fast-forward
 .github/workflows/cpu_build.yml                    |   8 -
 .github/workflows/docs_build.yml                   |   1 -
 .github/workflows/release_src_artifact.yml         |  39 +-
 .gitmodules                                        |  24 -
 CMakeLists.txt                                     |  59 +-
...
```

Build Environment
-----------------
Compiler version: MSVC 2017 v141
Operating system: Windows 10
Arrayfire version: d86edd18 was latest local commit when launching the pull.
git version: 2.23.0.windows.1
Build environment: NA
CMake variables: NA

**Comment by 9prady9** [comment]
@willyborn Hey, there have been some changes in our code base in terms of how we fetch certain dependencies(header only libraries, test data etc i.e. build-time dependencies). We have moved away from submodule approach to a cmake's based approach.

With latest master, there is no need to run `git submodule update --init --recursive` or even the need to clone `arrayfire` recursively. 

- The most simplest way to resolve this is to remove your clone and clone it freshly without `--recursive` flag.

- If you want to keep your current clone, then you would need to remove all the subdirectories that were submodules earlier.

- Another approach would be to, undo your pull if you can and the run `git submodule deinint -f .` that should remove all the needed submodule directories.

**Comment by 9prady9** [comment]
Also note that we have added a new `AF_BUILD_OFFLINE` cmake option should be interested in it. This option is intended to be used when all build-time dependencies that are fetched by cmake are already available under `<arrayfire-clone-path>/build/extern` path.

Another way to do such an offline build is to download our source tar balls that are attached to our releases on GitHub which have all these dependencies included in it. Please keep in mind that these changes will be reflected only from next fix release, v3.8.1

**Comment by 9prady9** [comment]
Closing since the question has been addressed and it is not a bug but rather due to change in code organization of source dependencies.
